### PR TITLE
Error handling for dbt retry

### LINKED ISSUE #3625: Add results examples to `dbt retry` page
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/commands/retry

### What part(s) of the page would you like to see updated?

Add examples of both successful and failed runs for `dbt retry`

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?
Added sample outputs to the `dbt retry` reference page

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/reference/commands/retry.md [hunk 1]
```diff
@@ -28,3 +28,79 @@ Retry works with the following commands:
 
 `dbt retry` reuses the [selectors](/reference/node-selection/yaml-selectors) from the previously executed command.
 
+
+If you run `dbt retry` after successfully executing `dbt run`, you'll get the following results.
```
Comment (matthewshaver) [inline website/docs/reference/commands/retry.md]:
```suggestion
Executing `dbt retry` after a successful `dbt run` will yield the following results:
```

File: website/docs/reference/commands/retry.md [hunk 2]
```diff
@@ -28,3 +28,79 @@ Retry works with the following commands:
...
+  
+Runtime Error in model customers (models/customers.sql)
+ Parser Error: syntax error at or near "selct"
+
+Done. PASS=4 WARN=0 ERROR=1 SKIP=0 TOTAL=5
+```
+
+
+Executing `dbt retry` without fixing the error(s) will result in another failed run:
+```
```
Comment (matthewshaver) [inline website/docs/reference/commands/retry.md]:
```suggestion

```
```

File: website/docs/reference/commands/retry.md [hunk 3]
```diff
@@ -28,3 +28,80 @@ Retry works with the following commands:
 
 `dbt retry` reuses the [selectors](/reference/node-selection/yaml-selectors) from the previously executed command.
 
+
+Executing `dbt retry` after a successful `dbt run` will yield the following results:
```
Comment (nghi-ly) [inline website/docs/reference/commands/retry.md]:
```suggestion
Example results of executing `dbt retry` after a successful `dbt run`:
```

File: website/docs/reference/commands/retry.md [hunk 4]
```diff
@@ -28,3 +28,80 @@ Retry works with the following commands:
...
+
+```
```
Comment (nghi-ly) [inline website/docs/reference/commands/retry.md]:
```suggestion
```shell
```

File: website/docs/reference/commands/retry.md [hunk 5]
```diff
@@ -28,3 +28,80 @@ Retry works with the following commands:
...
+Running with dbt=1.6.1
+Registered adapter: duckdb=1.6.0
+Found 5 models, 3 seeds, 20 tests, 0 sources, 0 exposures, 0 metrics, 348 macros, 0 groups, 0 semantic models
+ 
+Nothing to do. Try checking your model configs and model specification args
+```
+
+In a scenario where `dbt run` has failed due to a syntax error in a model:
```
Comment (nghi-ly) [inline website/docs/reference/commands/retry.md]:
```suggestion
Example of when `dbt run` encounters a syntax error in a model: 
```

File: website/docs/reference/commands/retry.md [hunk 6]
```diff
@@ -28,3 +28,80 @@ Retry works with the following commands:
...
+
+```
```
Comment (nghi-ly) [inline website/docs/reference/commands/retry.md]:
```suggestion
```shell
```

File: website/docs/reference/commands/retry.md [hunk 7]
```diff
@@ -28,3 +28,80 @@ Retry works with the following commands:
...
+Runtime Error in model customers (models/customers.sql)
+ Parser Error: syntax error at or near "selct"
+
+Done. PASS=4 WARN=0 ERROR=1 SKIP=0 TOTAL=5
+```
+
+
+Executing `dbt retry` without fixing the error(s) will result in another failed run:
+
+```
```
Comment (nghi-ly) [inline website/docs/reference/commands/retry.md]:
```suggestion
```shell
```

File: website/docs/reference/commands/retry.md [hunk 8]
```diff
@@ -28,3 +28,80 @@ Retry works with the following commands:
...
+Completed with 1 error and 0 warnings:
+  
+Runtime Error in model customers (models/customers.sql)
+ Parser Error: syntax error at or near "selct"
+
+Done. PASS=4 WARN=0 ERROR=1 SKIP=0 TOTAL=5
+```
+
+
+Executing `dbt retry` without fixing the error(s) will result in another failed run:
```
Comment (nghi-ly) [inline website/docs/reference/commands/retry.md]:
```suggestion
Example of a subsequent failed `dbt retry` run without fixing the error(s):
```

File: website/docs/reference/commands/retry.md [hunk 9]
```diff
@@ -28,3 +28,80 @@ Retry works with the following commands:
...
+
+1 of 1 START sql table model main.customers .................................... [RUN]
+1 of 1 ERROR creating sql table model main.customers ........................... [ERROR in 0.03s]
+
+Done. PASS=4 WARN=0 ERROR=1 SKIP=0 TOTAL=5
+```
+
+Executing `dbt retry` after a failed run, but correcting the issue, will result in a completed run:
+
+```
```
Comment (nghi-ly) [inline website/docs/reference/commands/retry.md]:
```suggestion
```shell
```

File: website/docs/reference/commands/retry.md [hunk 10]
```diff
@@ -28,3 +28,80 @@ Retry works with the following commands:
...
+
+Concurrency: 24 threads (target='dev')
+
+1 of 1 START sql table model main.customers .................................... [RUN]
+1 of 1 ERROR creating sql table model main.customers ........................... [ERROR in 0.03s]
+
+Done. PASS=4 WARN=0 ERROR=1 SKIP=0 TOTAL=5
+```
+
+Executing `dbt retry` after a failed run, but correcting the issue, will result in a completed run:
```
Comment (nghi-ly) [inline website/docs/reference/commands/retry.md]:
```suggestion
Example of a successful `dbt retry` run after fixing error(s):
```

**Review by nghi-ly — COMMENTED** [review]
thank you for adding the examples! left minor suggs for conciseness. for syntax highlighting, sugg adding language in the code blocks. can't recall how `shell` renders but is def colorful for `sql` and `python`

**Review by nghi-ly — APPROVED** [review]

**Pull request merged**